### PR TITLE
Retry specs in PR mode?

### PR DESCRIPTION
**Pull request by ppp0**
Because of network failures etc single specs can fail.
How about we retry specs X times during pull-request mode?

### CODE REVIEW

**Comment by ppp0** [comment]
@kris-lab please review

**Comment by kris-lab** [comment]
Maybe I am missing something, but what about combining `retries` and `no_retries`. It looks like `no_retries=false` is basically `retries=0`? Then the default would be `retries=0` and only for CI it would be positive?

**Comment by ppp0** [comment]
Fully agree, @kris-lab please re-run reviewing task

File: build-ci.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ bundle exec rake lint
 vagrant box update
 if [ -z "${ghprbTargetBranch}" ]; then
     # Full project build
-    bundle exec rake spec
+    retries=2 bundle exec rake spec
```
Comment (kris-lab) [inline build-ci.sh]: maybe `retries` should go to the end of line, I think it will be cleaner, wdyt?

File: spec/lib/spec_runner.rb [hunk 2]
```diff
@@ -189,7 +190,12 @@ def run
         spec.puppet_module.supported_os_list.each do |os|
           next unless @filter_os_list.nil? or @filter_os_list.include? os
           emit(:output, "Running #{spec.name} for #{os}\n".bold)
```
Comment (kris-lab) [inline spec/lib/spec_runner.rb]: This line seems to be repeated twice: on line 192 and line 197?

**Comment by kris-lab** [comment]
I have added small comments. The rest looks and works fine for me.

File: build-ci.sh (hunk 1 above)
Comment (ppp0) [inline build-ci.sh]: I've been having issues at times when programs do not recognise environment variables as such but rather think they're arguments. It depends on how the program reads it's ARGV list. That's the reason why I alwasy like to prepend ENV variables. OTOH, I believe `bundle` and `rake` will handle this properly. Wdty?

Comment (kris-lab) [inline build-ci.sh]: ok!

File: spec/lib/spec_runner.rb (hunk 2 above)
Comment (kris-lab) [inline spec/lib/spec_runner.rb]: ok

**Comment by kris-lab** [comment]
ok lgtm

**Pull request merged**